src/main/java/picard/illumina/CollectIlluminaBasecallingMetrics.java

```diff
@@ -178,38 +195,94 @@ protected int doWork() {
                         IlluminaDataType.PF,
                         IlluminaDataType.Position,
                         IlluminaDataType.Barcodes);
+            }
         }
 
         unmatched_barcode = StringUtil.repeatCharNTimes('N', barcodeLength);
 
         //Initialize data provider, iterate over clusters, and collect statistics
-        final BaseIlluminaDataProvider provider = factory.makeDataProvider();
+        //  BaseIlluminaDataProvider provider;
+        if (USE_NEW_CONVERTER) {
+            File laneDir = new File(BASECALLS_DIR, IlluminaFileUtil.longLaneStr(LANE));
+
+            File[] cycleDirs = IOUtil.getFilesMatchingRegexp(laneDir, IlluminaFileUtil.CYCLE_SUBDIRECTORY_PATTERN);
+
+            //CBCLs
+            List<File> cbcls = new ArrayList<>();
+            Arrays.asList(cycleDirs)
+                    .forEach(cycleDir -> cbcls.addAll(
+                            Arrays.asList(IOUtil.getFilesMatchingRegexp(
+                                    cycleDir, "^" + IlluminaFileUtil.longLaneStr(LANE) + "_(\\d{1,5}).cbcl$"))));
+
+            if (cbcls.size() == 0) {
+                throw new PicardException("No CBCL files found.");
+            }
+
+            IOUtil.assertFilesAreReadable(cbcls);
+
+            //locs
+            List<AbstractIlluminaPositionFileReader.PositionInfo> locs = new ArrayList<>();
+            File locsFile = new File(BASECALLS_DIR.getParentFile(), "s.locs");
```

Comment:
We should use a constant for this, e.g. `AbstractIlluminaPositionFileReader.S_LOCS_FILE`